Mathlib/Util/Term/Basic.lean

```diff
@@ -0,0 +1,47 @@
+/-
+Copyright (c) 2021 Gabriel Ebner. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gabriel Ebner, Yuyang Zhao
+-/
+import Lean.Elab.SyntheticMVars
+import Lean.Meta.Tactic.Delta
+import Mathlib.Lean.Expr.Basic
+import Mathlib.Util.Term.Beta
+
+/-!
+# Basic term elaborators
+-/
+
+namespace Mathlib.Util.Term.Basic
+
+open Lean Elab Term Meta
+
+/-- `delta% t` elaborates to a head-delta reduced version of `t`. -/
+elab "delta% " t:term : term <= expectedType => do
```

Comment:
The `<= expectedType` syntax causes postponement if the expected type is not known yet. Is that what we want?

The reason the `beta%` elaborator is written the way it is is to elaborate immediately.